### PR TITLE
refactor(consumables): callback

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

- Fix improper parameters on some of the natives
- Use ox_lib progressbar anims and props over scully emotes
- Use a callback for item usages instead of individual events

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

**Review by TonybynMp4 — APPROVED** [review]
goot change

File: client/consumables.lua [hunk 1]
```diff
@@ -13,18 +13,18 @@ local function healOxy()
     local count = 9
     while count > 0 do
         Wait(1000)
-        count -= 1
+        count = count - 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

File: client/consumables.lua [hunk 2]
```diff
@@ -52,7 +52,7 @@ local function ecstasyEffect()
     SetFlash(0, 0, 500, 7000, 500)
     while startStamina > 0 do
         Wait(1000)
-        startStamina -= 1
+        startStamina = startStamina - 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

File: client/consumables.lua [hunk 3]
```diff
@@ -84,7 +84,7 @@ local function crackBaggyEffect()
         if math.random(1, 100) < 10 then
             RestorePlayerStamina(cache.playerId, 1.0)
         end
-        startStamina -= 1
+        startStamina = startStamina - 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

File: client/consumables.lua [hunk 4]
```diff
@@ -107,7 +107,7 @@ local function cokeBaggyEffect()
         if math.random(1, 100) < 20 then
             RestorePlayerStamina(cache.playerId, 1.0)
         end
-        startStamina -= 1
+        startStamina = startStamina - 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

File: client/consumables.lua [hunk 5]
```diff
@@ -127,7 +127,7 @@ local function smokeWeed()
         while smokingWeed do
             Wait(10000)
             TriggerServerEvent('hud:server:RelieveStress', math.random(15, 18))
-            relieveCount += 1
+            relieveCount = relieveCount + 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

File: client/consumables.lua [hunk 6]
```diff
@@ -194,21 +223,33 @@ RegisterNetEvent('consumables:client:DrinkAlcohol', function(itemName)
...
         }
     }) then -- if completed
-        exports.scully_emotemenu:cancelEmote()
-        TriggerEvent('inventory:client:ItemBox', exports.ox_inventory:Items()[itemName], 'remove')
-        TriggerServerEvent('consumables:server:drinkAlcohol', itemName)
-        TriggerServerEvent('consumables:server:addThirst', QBX.PlayerData.metadata.thirst + ConsumablesAlcohol[itemName])
+        local used = lib.callback.await('consumables:server:usedItem', false, itemName)
+        if not used then return end
+
+        TriggerServerEvent('consumables:server:addThirst', {name = itemName,amount = QBX.PlayerData.metadata.thirst + ConsumablesAlcohol[itemName]})
```
Comment (BerkieBb) [inline client/consumables.lua]: Spaces are allowed between comma's

File: client/consumables.lua [hunk 7]
```diff
@@ -194,21 +223,33 @@ RegisterNetEvent('consumables:client:DrinkAlcohol', function(itemName)
...
         TriggerServerEvent('hud:server:RelieveStress', math.random(2, 4))
-        alcoholCount += 1
+        alcoholCount = alcoholCount + 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

File: client/consumables.lua [hunk 8]
```diff
@@ -375,7 +425,7 @@ CreateThread(function()
         Wait(10)
         if alcoholCount > 0 then
             Wait(1000 * 60 * 15)
-            alcoholCount -= 1
+            alcoholCount = alcoholCount - 1
```
Comment (BerkieBb) [inline client/consumables.lua]: ?

**Review by BerkieBb — APPROVED** [review]

**Pull request merged**